WordPress/Classes/Utility/Logging/SentryStartupEvent.swift

```diff
@@ -48,9 +45,33 @@ startup time. This will block the thread. Do not use unless you're sure.
             ]
         })
 
-        client.send(event: event, completion: { _ in
+        CrashLogging.logErrorAndWait(NSError(domain: title, code: -1, userInfo: [NSLocalizedDescriptionKey: title]), userInfo: userInfo)
+    }
+}
+
+extension CrashLogging {
+    /**
+     Writes the error to the Crash Logging system, and includes a stack trace. This method will block the thread until the event is fired.
+
+     - Parameters:
+     - error: The error object
+     - userInfo: A dictionary containing additional data about this error.
+     - level: The level of severity to report in Sentry (`.error` by default)
+    */
+    static func logErrorAndWait(_ error: Error, userInfo: [String: Any]? = nil, level: SentrySeverity = .error) {
+        let event = Event(level: .error)
+        event.message = error.localizedDescription
+        event.extra = userInfo ?? (error as NSError).userInfo
+
+        Client.shared?.snapshotStacktrace {
+            Client.shared?.appendStacktrace(to: event)
+        }
+
+        let semaphore = DispatchSemaphore(value: 0)
+
+        Client.shared?.send(event: event) { _ in
             semaphore.signal()
-        })
+        }
 
         semaphore.wait()
```

Comment:
If `Client.shared` is nil, this line could cause a deadlock. WDYT if we bail out if the `Client.shared` is nil? (We can call `fatalError`, not sure who sets the Client, or when!)

```
guard let client = Client.shared else {
   fatalError() 
}
```

Reply:
I just used the same logic and kept that way for consistency, ref: https://github.com/Automattic/Automattic-Tracks-iOS/blob/develop/Automattic-Tracks-iOS/Crash%20Logging/CrashLogging.swift#L130

(Since I'm going to move this method to there)

Reply:
This one is sliiiightly (slightly) different because of the Semaphore. A nil in there could result in a deadlock (no idea if that could ever happen, though!), but it wouldn't generate a crashlog. I think!

An in between approach would maybe be... using a force unwrap when calling `Client.shared`, so that worst case scenario we'd get a crashlog. WDYT?

Reply:
Oooh ok, I see your point now!

Makes sense, done in b41c4d5fa71090775442ba78f21dd65121861998

I just changed it slightly: instead of crashing, if the Client doesn't exist the method return and then the execution will follow (and probably it will crash on the original crash).

Reply:
Thanks for doing this @leandroalonso !!